tools/src/main/java/io/narayana/db/DBAllocator.java

```diff
@@ -0,0 +1,328 @@
+        }
+        LOGGER.info(String.format("Deallocate DB returned HTTP status code: %d", deallocstatusCode));
+        return deallocstatusCode == 200;
+    }
+
+    public boolean reallocateDB(final int expiryMinutes, final DB db) {
+        // Something between nothing and a business week...
+        if (expiryMinutes < 0 || expiryMinutes > 7200) {
+            throw new IllegalArgumentException("expiryMinutes out of range 0 - 7200");
+        }
+        final String dballocatorHostPort = System.getProperty("dballocator.host.port");
+        if (StringUtils.isEmpty(dballocatorHostPort)) {
+            throw new IllegalArgumentException("dballocator.host.port must not be empty");
+        }
+        if (db.allocationProperties == null) {
+            throw new IllegalArgumentException("db.allocationProperties must not be null");
+        }
+        if (StringUtils.isEmpty(db.allocationProperties.getProperty("uuid"))) {
+            throw new IllegalArgumentException("uuid must not be empty");
+        }
+        final URL realloc;
+        try {
+            realloc = new URL(String.format("%s/Allocator/AllocatorServlet?operation=realloc&uuid=%s&expiry=%s",
+                    dballocatorHostPort,
+                    db.allocationProperties.getProperty("uuid"),
+                    System.getProperty("dballocator.expiry.minutes")
+            ));
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+            LOGGER.severe("Failed to construct reallocate DB URL, see: " + e.getMessage());
+            return false;
+        }
+        LOGGER.info("Reallocating database...");
+        final int reallocstatusCode;
+        try {
+            reallocstatusCode = ((HttpURLConnection) realloc.openConnection()).getResponseCode();
+        } catch (IOException e) {
+            e.printStackTrace();
+            LOGGER.severe("Failed to reallocate DB, see: " + e.getMessage());
+            return false;
+        }
+        LOGGER.info(String.format("Reallocate DB returned HTTP status code: %d", reallocstatusCode));
+        return reallocstatusCode == 200;
+    }
+
+    public boolean reallocateDB(final DB db) {
+        if (db == null) {
+            throw new IllegalArgumentException("db must not be null");
+        }
+        if (StringUtils.isEmpty(System.getProperty("dballocator.expiry.minutes"))) {
+            throw new IllegalArgumentException("dballocator.expiry.minutes must not be empty");
+        }
+        return reallocateDB(Integer.parseInt(System.getProperty("dballocator.expiry.minutes")), db);
+    }
+
+    public boolean cleanDB(final DB db) {
+        if (db == null) {
+            throw new IllegalArgumentException("db must not be null");
+        }
+        if (db.allocationProperties == null) {
+            throw new IllegalArgumentException("db.allocationProperties must not be null");
+        }
+        final String dballocatorHostPort = System.getProperty("dballocator.host.port");
```

Comment:
would not be benefical to create some utility method which get `property name` and returns value or throws `IAE` if empty? I think it could remove quite a lot of `if` causes from the start of all the methods around.